test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperBlockRewriterTest.cs

```diff
@@ -3029,10 +3086,85 @@ public static TheoryData MinimizedAttributeData_CSharpBlock
+                                            chunkGenerator.Prefix.Location.LineIndex,
+                                            chunkGenerator.Prefix.Location.CharacterIndex + 2)),
+                                    new SourceLocation(
+                                        chunkGenerator.ValueStart.AbsoluteIndex + 2,
+                                        chunkGenerator.ValueStart.LineIndex,
+                                        chunkGenerator.ValueStart.CharacterIndex + 2));
+
+                                expressionBlockBuilder.ChunkGenerator = newChunkGenerator;
+                                blockBuilder.Children[0] = expressionBlockBuilder.Build();
+
+                                for (var j = 1; j < blockBuilder.Children.Count; j++)
+                                {
+                                    var span = blockBuilder.Children[j] as Span;
+                                    if (span != null)
+                                    {
+                                        var literalChunkGenerator =
+                                            span.ChunkGenerator as LiteralAttributeChunkGenerator;
+
+                                        var spanBuilder = new SpanBuilder(span);
+                                        spanBuilder.ChunkGenerator = new LiteralAttributeChunkGenerator(
+                                            prefix: new LocationTagged<string>(
+                                                literalChunkGenerator.Prefix.Value,
+                                                new SourceLocation(
+                                                    literalChunkGenerator.Prefix.Location.AbsoluteIndex + 2,
+                                                    literalChunkGenerator.Prefix.Location.LineIndex,
+                                                    literalChunkGenerator.Prefix.Location.CharacterIndex + 2)),
+                                            value: new LocationTagged<string>(
+                                                literalChunkGenerator.Value.Value,
+                                                new SourceLocation(
+                                                    literalChunkGenerator.Value.Location.AbsoluteIndex + 2,
+                                                    literalChunkGenerator.Value.Location.LineIndex,
+                                                    literalChunkGenerator.Value.Location.CharacterIndex + 2)));
+
+                                        blockBuilder.Children[j] = spanBuilder.Build();
+                                    }
+                                }
+
+                                tagHelperBlock.Attributes[i] = new KeyValuePair<string, SyntaxTreeNode>(
+                                    attribute.Key,
+                                    blockBuilder.Build());
+                            }
+                        }
+                    }
+                };
```

Comment:
We no longer rewrite `LiteralAttributeChunkGenerator's and `DynamicAttributeChunkGenerator`s for unbound `TagHelper` attributes. Sadly they both require source locations :cry: